src/pudl_rmi/validate.py

```diff
@@ -0,0 +1,178 @@
+        )
+    )
+    npb.loc[:, "ferc_acct_name"] = npb.ferc_acct_name.str.title()
+    return npb
+
+
+def add_data_source(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a data_source column with 'FERC' and 'PUC'.
+
+    The depreciation data includes a `data_source` column as a part of the
+    primary key of that data set. This is because there are sometimes we have
+    depreciation study from both `FERC` and `PUC` for the same utility and
+    plants. Because of this, in order to compare the FERC plant input tables to
+    the FERC-EIA-Depreciation tables we need to add *both* versions of the
+    depreciation `data_source`.
+    """
+    if "data_source" not in df:
+        df = pd.concat(
+            [
+                df.assign(data_source="PUC"),
+                df.assign(data_source="FERC"),
+            ]
+        )
+    return df
+
+
+def group_sum_cols(df, data_cols: List[str], by: List[str]) -> pd.DataFrame:
+    """Groupby sum a specific table's data cols."""
+    # convert date to year bc many of the og depish studies are EOY
+    summed_out = (
+        df.assign(report_year=lambda x: x.report_date.dt.year)
+        .astype({"report_year": pd.Int64Dtype()})
+        .groupby(by=by, dropna=True)[data_cols]
+        .sum(min_count=1)
+    )
+    return summed_out
+
+
+def agg_test_data(
+    df1: pd.DataFrame, df2: pd.DataFrame, data_cols: List[str], by: List[str], **kwarg
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column are equal.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``.
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_cols: data columns to check. Columns must be in both ``df1`` and
+            ``df2``.
+    """
+    test = pd.merge(
+        group_sum_cols(df1.pipe(add_data_source), data_cols=data_cols, by=by),
+        group_sum_cols(df2.pipe(add_data_source), data_cols=data_cols, by=by),
+        right_index=True,
+        left_index=True,
+        suffixes=("_1", "_2"),
+        how="outer",
+    ).astype("float64")
+    for data_col in data_cols:
+        test.loc[:, f"{data_col}_isclose"] = np.isclose(
+            test[f"{data_col}_1"],
+            test[f"{data_col}_2"],
+            equal_nan=True,
+            **kwarg,
+        )
+        test.loc[:, f"{data_col}_ratio"] = test[f"{data_col}_1"] / test[f"{data_col}_2"]
+        logger.info(
+            f"{data_col} close records: {len(test[test[f'{data_col}_isclose']])/len(test):.02%}"
+        )
+    return test.sort_index()
+
+
+def test_df_vs_net_plant_balance(
+    ferc_deprish_eia: pd.DataFrame,
+    pudl_engine: sa.engine.Engine,
+    data_cols: List = [
+        "plant_balance_w_common",
+        "book_reserve_w_common",
+        "unaccrued_balance_w_common",
+    ],
+    rtol: float = 5e-02,
+    atol: float = 5e-02,
+) -> pd.DataFrame:
+    """
+    Compare FERC-Deprish-EIA output to FERC1 Net Plant Balance totals.
```

Comment:
Compare and... then what? Just collate the data for later inspection? Or raise an exception?

Reply:
I think maybe calling this function `compare_` rather than `test_` or `validate_` might make be helpful. Does `df` here stand for DataFrame? Or `deprish_ferc1`? It's still unclear from the docstring what the function is doing -- what "compare" means. It seems like you're doing 2 things for each of the `data_cols`:
* Adding a boolean column that indicates whether the `ferc1_deprish` and `npb` versions of the aggregated `data_col` are sufficiently similar (based on the values of `rtol` and `atol`).
* Calculating the ratio between the aggregated `data_col` values so the user can see how similar or different they are to each other.

Reply:
`compare_` sounds great to me!

`df` here stands for dataframe. one could run this comparison with many of the different "stages" of this data. I expect to use it mostly with the `deprish_ferc1` data (bc that is the "end of the line")... but as with the integration tests, I was trying to enable this to be run with many different inputs.

and yes to this description... I grabbed it and put it into the docs

Reply:
Given that the arguments to the function are named for specific tables, not generic dataframes, using `df` in the function name seems unintuitive. I think either make it generic or specific in both the function name, and the argument names.